src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -162,7 +161,29 @@ private CachedCoinView(IDateTimeProvider dateTimeProvider, ILoggerFactory logger
         }
 
         /// <inheritdoc />
-        public override async Task<FetchCoinsResponse> FetchCoinsAsync(uint256[] txIds, CancellationToken cancellationToken = default(CancellationToken))
+        public Task<uint256> GetTipHashAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            Task<uint256> task = Task.Run(async () =>
```

Comment:
Why this wrapper?

Reply:
Just because in this class we use wrappers everywhere. Would you like me to remove it?

Reply:
Removed wrapper